core/src/main/scala/org/ensime/indexer/ClassfileIndexer.scala

```diff
@@ -65,29 +63,54 @@ final class ClassfileIndexer(file: FileObject) extends SLF4JLogging {
         None
       }
 
+      val interfaceNames: List[ClassName] = interfaces.map(ClassName.fromInternal)(collection.breakOut)
+      val superClass = Option(superName).map(ClassName.fromInternal)
+
+      internalRefs.addAll(interfaceNames.asJava)
+      internalRefs.addAll(superClass.toList.asJava)
+
       clazz = RawClassfile(
         ClassName.fromInternal(name),
         signatureClass,
-        Option(superName).map(ClassName.fromInternal),
-        interfaces.map(ClassName.fromInternal)(breakOut),
+        Set.empty,
+        superClass,
+        interfaceNames,
         Access(access),
         (ACC_DEPRECATED & access) > 0,
-        Queue.empty, Queue.empty, RawSource(None, None)
+        Nil, Queue.empty, RawSource(None, None),
+        isScala = false,
+        internalRefs.asScala
```

Comment:
it will be quicker for this to be converted into a `List` (we'd be relying on convention that it's actually distinct)